units/src/amount/error.rs

```diff
@@ -389,3 +390,44 @@ impl fmt::Display for PossiblyConfusingDenominationError {
 impl std::error::Error for PossiblyConfusingDenominationError {
     fn source(&self) -> Option<&(dyn std::error::Error + 'static)> { None }
 }
+
+/// An error consensus decoding an `Amount`.
+#[cfg(feature = "encoding")]
+#[derive(Debug, Clone, PartialEq, Eq)]
+#[non_exhaustive]
+pub enum AmountDecodeError {
```

Comment:
As far as I understood the discussion it should be `AmountDecoderError` (note the 'r' in decoder). I.e., its the name of the decoder type with `Error` suffixed.